### PR TITLE
Move odo to the umbrella

### PR DESCRIPTION
**Pull request by LtChae**
This move has negated the need for the odo integration tests. They have either been incorporated into the e2e test or downgraded to unit tests.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -25,3 +25,5 @@ erl_crash.dump
 
 node_modules/
 .DS_Store
+test/support/minio_data/.minio.sys/*
```
Comment (bbrewer-papa-x) [inline .gitignore]: i think you can safely delete these lines now that we've changed the way that we pre-load data into the minio container image during testing.

File: apps/odo/.formatter.exs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Used by "mix format"
```
Comment (bbrewer-papa-x) [inline apps/odo/.formatter.exs]: do we have a formatter for each app in the apps directory or can we get away with a single top-level one?

File: apps/odo/.gitignore [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# The directory Mix will write compiled artifacts to.
```
Comment (bbrewer-papa-x) [inline apps/odo/.gitignore]: i think we can safely remove this sub-gitignore

File: apps/odo/config/config.exs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+use Mix.Config
+
+config :odo,
+  collector: StreamingMetrics.PrometheusMetricCollector
+
+config :mime, :types, %{
```
Comment (bbrewer-papa-x) [inline apps/odo/config/config.exs]: this `mime` block should be removed now as it will inherit from the top-level config file in the umbrella.

File: apps/odo/mix.exs [hunk 5]
```diff
@@ -0,0 +1,62 @@
+defmodule Odo.MixProject do
+  use Mix.Project
+
+  def project do
+    [
+      app: :odo,
+      version: "0.2.0",
```
Comment (bbrewer-papa-x) [inline apps/odo/mix.exs]: might want to check and make sure we don't already have a release tagged with `0.2.0` and if we do, rev this slightly.

File: apps/odo/test/unit/odo/angry_case_test.exs [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  import SmartCity.Event, only: [file_ingest_end: 0, error_file_ingest: 0]
+  alias ExAws.S3
+  alias SmartCity.HostedFile
+
+  test "Individual task failures do not stop others" do
+    start_supervised!(Odo.TestEventHandler)
+
+    # Application.put_env(:odo, :working_dir, "test/support")
+
+    File.cp!("test/support/my-data.shapefile", "/tmp/123.shapefile")
```
Comment (bbrewer-papa-x) [inline apps/odo/test/unit/odo/angry_case_test.exs]: do this tempdir work without using the `Temp` module to create a platform-independent tempdir that the app has permissions to write to? we had issues with trying to write to `/tmp` when we were moving the integration test to the e2e integration.

File: apps/odo/test/unit/odo/angry_case_test.exs [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+      type: "error:file:ingest"
+    }
+
+    allow(ExAws.request(any()), return: {:ok, :there})
+
+    Brook.Event.send(Odo.event_stream_instance(), file_ingest_end(), :odo, bad_event)
+    Brook.Event.send(Odo.event_stream_instance(), file_ingest_end(), :odo, good_event)
+
+    SmartCity.TestHelper.eventually(fn ->
+      events = Odo.TestEventHandler.get_events()
```
Comment (bbrewer-papa-x) [inline apps/odo/test/unit/odo/angry_case_test.exs]: Is there a reason we don't just use `assert_receive` here like several other tests in the umbrella already do?

Comment (bbrewer-papa-x) [reply apps/odo/test/unit/odo/angry_case_test.exs]: Ex. in `reaper`'s handler_test it does a

```elixir
      assert_receive {:brook_event, %Brook.Event{type: "data:ingest:start", data: dataset}}
```

This seems slightly more straightforward to me at least.

Comment (bbrewer-papa-x) [reply apps/odo/test/unit/odo/angry_case_test.exs]: Balser/a person needs to update the Brook docs :/

The Test driver will send those `:brook_event` messages. I had to search the repo to confirm this https://github.com/bbalser/brook/search?q=%3Abrook_event&unscoped_q=%3Abrook_event

File: apps/odo/.formatter.exs (hunk 2 above)
Comment (LtChae) [reply apps/odo/.formatter.exs]: We have one per app. Not sure if we can do a top level one or not.

File: apps/odo/test/unit/odo/angry_case_test.exs (hunk 7 above)
Comment (LtChae) [reply apps/odo/test/unit/odo/angry_case_test.exs]: We decided to stick with the current implementation. We were not able to use assert_received because the test driver does not propagate events. It requires the test to use `Brook.Test.send`, but the event handler creates new events using `Brook.Event.send`, which is not picked up by the test driver.

File: apps/odo/test/unit/odo/angry_case_test.exs (hunk 6 above)
Comment (LtChae) [reply apps/odo/test/unit/odo/angry_case_test.exs]: Moved to use the Temp library

File: apps/odo/config/config.exs (hunk 4 above)
Comment (LtChae) [reply apps/odo/config/config.exs]: Removed

File: apps/odo/mix.exs (hunk 5 above)
Comment (LtChae) [reply apps/odo/mix.exs]: Bumped to 0.3.0

File: apps/odo/.gitignore (hunk 3 above)
Comment (LtChae) [reply apps/odo/.gitignore]: Removed

File: apps/estuary/mix.exs [hunk 8]
```diff
@@ -19,15 +19,13 @@ defmodule Estuary.MixProject do
     ]
   end
 
-  # Run "mix help compile.app" to learn about applications.
```
Comment (ManApart) [inline apps/estuary/mix.exs]: Thanks for getting rid of these boiler plate comments.

File: apps/odo/config/integration.exs [hunk 9]
```diff
@@ -0,0 +1,64 @@
+use Mix.Config
+# We should turn this into a repo to share with Reaper
```
Comment (ManApart) [inline apps/odo/config/integration.exs]: ^ Do we get this for free now that they're all umbrella?

File: apps/odo/lib/odo.ex [hunk 10]
```diff
@@ -0,0 +1,10 @@
+defmodule Odo do
+  @moduledoc """
+  Convenience functions for the Odo application.
+  """
+
+  def event_stream_instance() do
```
Comment (ManApart) [inline apps/odo/lib/odo.ex]: It'd be nice if this was closer to our convention in other apps (just `instance()`) but  I understand if that feels out of scope for this card.

Comment (jeffgrunewald) [reply apps/odo/lib/odo.ex]: this was more or less a backwards compatibility issue. We created reaper's `instance()` function (or equivalent) to return the atom `:reaper_brook` but now we have other non-Brook use cases for an `instance`-type function and we didn't want to break the existing view state by renaming that instance return value when we have a live system using it.

File: apps/odo/config/integration.exs (hunk 9 above)
Comment (jeffgrunewald) [reply apps/odo/config/integration.exs]: yes but that block should theoretically be able to be removed now that we don't have any integration tests for Odo (they should all be covered by unit tests or e2e).

File: apps/odo/.formatter.exs (hunk 2 above)
Comment (jeffgrunewald) [reply apps/odo/.formatter.exs]: we should experiment if those different files will conflict with each other. If they have different values (like line length) and you run the mix command at one level, i suspect you could be undoing the work the formatter did when you ran it at the top level and vice versa.
